contracts/contracts/FleekAccessControl.sol

```diff
@@ -5,227 +5,178 @@ pragma solidity ^0.8.7;
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 
+error MustHaveCollectionRole(uint8 role);
+error MustHaveTokenRole(uint256 tokenId, uint8 role);
+error MustHaveAtLeastOneOwner();
+error RoleAlreadySet();
+
 contract FleekAccessControl is Initializable {
     using Counters for Counters.Counter;
 
-    enum Roles {
-        Owner,
-        Controller
+    /**
+     * @dev All available collection roles.
+     */
+    enum CollectionRoles {
```

Comment:
Since it only has one value, maybe we can use just a string for this to make the tests more readable, too.

Reply:
The thing is that it may not be a single value in the future as we add new roles. For example the verifier role we have been discussing.

Reply:
that's true. for now, I think string looks better but if we are planning on adding more roles soon, then let's keep it this way.